src/biconnectivity/bridge.jl

```diff
@@ -0,0 +1,75 @@
+"""
+    Bridges{T}
+A state type for the depth-first search that finds bridges in a graph.
+"""
+mutable struct Bridges{T<:Integer}
+    low::Vector{T}
+    depth::Vector{T}
+    bridges::Vector{Edge{T}}
+    id::T
+end
+
+function Bridges(g::AbstractGraph)
+    n = nv(g)
+    T = typeof(n)
+    return Bridges(zeros(T, n), zeros(T, n), Edge{T}[], zero(T))
+end
+
+"""
+    visit!(state, g, u, v)
+Perform a depth first search storing the depth (in `depth`) and low-points
+(in `low`) of each vertex.
+"""
+function visit!(state::Bridges, g::AbstractGraph, u::Integer, v::Integer)
+    state.id += 1
+    state.depth[v] = state.id
+    state.low[v] = state.depth[v]
+
+    for w in outneighbors(g, v)
+        if state.depth[w] == 0
+            visit!(state, g, v, w)
+
+            state.low[v] = min(state.low[v], state.low[w])
+            if state.low[w] > state.depth[v]
+                edge = v < w ? Edge(v, w) : Edge(w, v)
+                push!(state.bridges, edge)
+            end
+
+        elseif w != u
+            state.low[v] = min(state.low[v], state.depth[w])
+        end
+    end
+end
+
+"""
+    bridge(g)
+Compute the [bridges](https://en.m.wikipedia.org/wiki/Bridge_(graph_theory))
+of a connected graph `g` and return an array containing all bridges.
+# Examples
+```jldoctest
+julia> using LightGraphs
+
+julia> bridge(StarGraph(5))
+8-element Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1}:
+ Edge 1 => 2
+ Edge 1 => 3
+ Edge 1 => 4
+ Edge 1 => 5
+
+julia> bridge(PathGraph(5))
+8-element Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1}:
+ Edge 4 => 5
+ Edge 3 => 4
+ Edge 2 => 3
+ Edge 1 => 2
+```
+"""
+function bridge(g::AbstractGraph)
```

Comment:
`bridge` is a bit ambiguous for directed graphs, better to restrict the dispatch.

Concerning the name, I agree `bridges` makes more sense. I wanted to be consistent with `articulation` (which should maybe be `articulations` instead?), but maybe it's better to call this `bridges` and  deprecate `articulation` to `articulations` in the future.

Reply:
I think `articulation` is shorthand for `articulation_points`. But I agree that this should be `bridges`.

Reply:
I see, yes this should definitely be `bridges`.

I'm a bit unsure what's the correct dispatch, because it is not clear to me that all graphs types defined in various packages are subtypes of `SimpleGraphs.AbstractSimpleGraph`. Maybe easier to accept all `AbstractGraph{<:Integer}` and throw an argument error if the graph is directed.

Reply:
I could also make the code a bit more general and accept abstract graph of non-integer eltype, though I'm not sure how common they are.